client/Scenes/Game_Menus/Login_Menu/Scripts/button_ui_connect.gd

```diff
@@ -1,8 +1,8 @@
 extends Button
 
 func _pressed():
-	manager_connection.HOST = get_parent().get_node("txt_IP").text
-	manager_connection.PORT = get_parent().get_node("txt_PORT").text
-	manager_connection.username = get_parent().get_node("txt_username").text
-	manager_connection.password = get_parent().get_node("txt_password").text
+	manager_connection.HOST = get_parent().get_parent().get_node("server_container/ip-port_container/txt_IP").text
```

Comment:
Rather than making this dependent on the relative hierarchy between the button and the text boxes, can you get these via an absolute file path?  Or even better, store a reference to the text boxes and pass it to this code.

The way it is now, these lines of code would need updating every time the hierarchical distance between the text boxes and button is changed.

Reply:
I know the absolute file path is easily possible.

Not sure about passing a reference though, but it sounds possible to me.
Just a bit out of my skill level.

Reply:
Fixed by commit https://github.com/stolencatkarma/CataclysmLD/pull/29/commits/db3e324a846d9184ba6fdef077f7c6b0cd3d71f4